### ISSUE TITLE
Monitor not sending utm parameters to GA

### ISSUE THREAD

**Issue by hoosteeno** [mozilla/blurts-server#1265]
When you visit monitor.firefox.com with a set of arbitrary UTM parameters, those parameters are not sent to GA servers for the monitor property. This impacts our ability to measure impact of e.g. in-product or other upstream links. 

STR:

* Visit https://monitor.firefox.com/?utm_source=test&utm_campaign=test&utm_content=test&utm_medium=referral
a) Look for realtime traffic from that source -- it won't appear: https://analytics.google.com/analytics/web/#/realtime/rt-traffic/a77033033w177898028p180612539/filter.list=14==REFERRAL;7~~=test;&metric.type=0/
b) Use [Google's Analytics Debugger](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna) to inspect the contents of the GA payload sent (see devtools console). It includes the host name, but not the parameters:

```
Running analytics_debug.js. This script is intended for testing and debugging only.
log @ analytics_debug.js:16
analytics_debug.js:16 Initializing Google Analytics.
analytics_debug.js:16 Running command: ga("create", "UA-77033033-16")
analytics_debug.js:16 Creating new tracker: t0
analytics_debug.js:16 Running command: ga("set", "anonymizeIp", true)
analytics_debug.js:16 Running command: ga("set", "transport", "beacon")
analytics_debug.js:16 Running command: ga("send", "pageview")
analytics_debug.js:16 Sending hit with transport method "beacon"
analytics_debug.js:16 
Sent beacon:
v=1&_v=j79d&aip=1&a=100790930&t=pageview&_s=1&dl=https%3A%2F%2Fmonitor.firefox.com%2F&ul=en-us&de=UTF-8&dt=Firefox%20Monitor&sd=24-bit&sr=1680x1050&vp=1506x851&je=0&_u=QACAAAAB~&jid=&gjid=&cid=138610038.1568737118&tid=UA-77033033-16&_gid=607632993.1570650957&z=1767132027


analytics_debug.js:16 _j1              (&jid)  
analytics_debug.js:16 _j2              (&gjid) 
analytics_debug.js:16 adSenseId        (&a)    100790930
analytics_debug.js:16 anonymizeIp      (&aip)  1
analytics_debug.js:16 apiVersion       (&v)    1
analytics_debug.js:16 clientId         (&cid)  138610038.1568737118
analytics_debug.js:16 encoding         (&de)   UTF-8
analytics_debug.js:16 hitType          (&t)    pageview
analytics_debug.js:16 javaEnabled      (&je)   0
analytics_debug.js:16 language         (&ul)   en-us
analytics_debug.js:16 location         (&dl)   https://monitor.firefox.com/
analytics_debug.js:16 screenColors     (&sd)   24-bit
analytics_debug.js:16 screenResolution (&sr)   1680x1050
analytics_debug.js:16 title            (&dt)   Firefox Monitor
analytics_debug.js:16 trackingId       (&tid)  UA-77033033-16
analytics_debug.js:16 viewportSize     (&vp)   1506x851
```

Compare the above to the debugger's output from e.g. https://www.mozilla.com/?utm_source=test&utm_campaign=test&utm_content=test&utm_medium=referral. The www.mozilla.org GA payload includes `location: https://www.mozilla.org/en-US/firefox/?utm_campaign=test&utm_content=test&utm_medium=referral&utm_source=test`.

**Comment by lesleyjanenorton** [comment]
@hoosteeno https://analytics.google.com/analytics/web/#/report/trafficsources-all-traffic/a77033033w177898028p180612539/  I am not sure what is going on with realtime but from the page I linked it looks like we are sending UTM parameters to GA.

**Comment by hoosteeno** [comment]
I'm not so sure. Most referrals that come with utms include a utm_campaign parameter. In Monitor's GA, [the "all campaigns" view shows that campaign traffic dropped to effectively zero](https://analytics.google.com/analytics/web/#/report/trafficsources-campaigns/a77033033w177898028p180612539/) on Sept. 19:

<img width="871" alt="Screen Shot 2019-10-12 at 2 25 50 PM" src="https://user-images.githubusercontent.com/94519/66707353-5b593f00-ecfc-11e9-9bc0-a2006be54e30.png">

And [in the source/medium view, direct traffic grew 10x](https://analytics.google.com/analytics/web/#/report/trafficsources-all-traffic/a77033033w177898028p180612539/explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.sourceMedium:(direct)%20~2F%20(none)), starting Sept. 19:

<img width="943" alt="Screen Shot 2019-10-12 at 2 29 07 PM" src="https://user-images.githubusercontent.com/94519/66707381-dc183b00-ecfc-11e9-966a-41fa2d72c039.png">

The jump in direct-source traffic happens on the same day as a complete cratering of various sources. For example, [emails](https://analytics.google.com/analytics/web/#/report/trafficsources-all-traffic/a77033033w177898028p180612539/_r.drilldown=analytics.sourceMedium:fx-monitor-emails%20~2F%20email):

<img width="932" alt="Screen Shot 2019-10-12 at 2 32 05 PM" src="https://user-images.githubusercontent.com/94519/66707407-33b6a680-ecfd-11e9-81a9-3ab6b974d2fc.png">

Or [Activity Stream](https://analytics.google.com/analytics/web/#/report/trafficsources-all-traffic/a77033033w177898028p180612539/_r.drilldown=analytics.sourceMedium:activity-stream%20~2F%20referral&explorer-table-dataTable.sortColumnName=analytics.sourceMedium&explorer-table-dataTable.sortDescending=true):

<img width="935" alt="Screen Shot 2019-10-12 at 2 34 37 PM" src="https://user-images.githubusercontent.com/94519/66707427-79736f00-ecfd-11e9-9b47-ab0b9b38b08a.png">

All of these are consistent with the behavior observed in GA's analytics debugger plugin: GA is not getting UTMs. In the chart you sent, after 2019-09-19, it's extracting the sources it can get from the HTTP Referer header.

I tried to set up a local dev environment to test out my theory for why this is happening, but the dev build is failing on my machine. The server keeps crashing[0]. 

My theory is that [GA's pageview send](https://github.com/mozilla/blurts-server/blob/change-password-top-10-breached-sites-1257/public/js/analytics_tracking_protection.js#L63) is deferred until after `location.pathname` is stripped of `location.search`. In which case, maybe something like `ga("set", "location", pageValue);` around [line 56](https://github.com/mozilla/blurts-server/blob/change-password-top-10-breached-sites-1257/public/js/analytics_tracking_protection.js#L56) would fix it?

[0]
Here's the error I'm getting. If you know how to get past it, I'd be happy to try to PR a fix for this.

```
[nodemon] starting `node server.js`
/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/dsn.js:66
            throw new error_1.SentryError('Invalid Dsn');
            ^

SentryError: Invalid Dsn
    at new SentryError (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/error.js:21:28)
    at Dsn.fromString (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/dsn.js:66:19)
    at new Dsn (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/dsn.js:39:18)
    at NodeClient.BaseClient (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/baseclient.js:113:24)
    at new NodeClient (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/node/dist/client.js:98:23)
    at Object.initAndBind (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/core/dist/sdk.js:20:18)
    at Object.init (/Users/username-removed/Documents/mozilla/projects/blurts-server/node_modules/@sentry/node/dist/sdk.js:74:12)
    at Object.<anonymous> (/Users/username-removed/Documents/mozilla/projects/blurts-server/server.js:6:8)
    at Module._compile (internal/modules/cjs/loader.js:945:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:962:10) {
  name: 'SentryError'
}
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by pdehaan** [comment]
@hoosteeno re:sentry error,
sounds similar to the recent #1268 pr.

**Comment by lesleyjanenorton** [comment]
@hoosteeno  Thank you for the right up. I think you're right. I can't find a single example of a utm_source/content/campaign/anything being sent to GA locally or on production. I'm happy to send up a PR for this tomorrow/later this week but will follow up on Slack regarding running this locally in any case.

**Comment by ddurst** [comment]
@hoosteeno Would you be so kind as to confirm this fix after today's push with some more purty graphics?

**Comment by hoosteeno** [comment]
Yes, things are looking up again. Thank you.

<img width="876" alt="Screen Shot 2019-10-18 at 1 52 11 PM" src="https://user-images.githubusercontent.com/94519/67124082-ff386400-f1ae-11e9-9a3c-73ff122a52e3.png">
